docs/pages/experiments/material-ui/divider.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import {
+  Experimental_CssVarsProvider as CssVarsProvider,
+  useColorScheme,
+} from '@mui/material/styles';
+import CssBaseline from '@mui/material/CssBaseline';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Divider from '@mui/material/Divider';
+import Moon from '@mui/icons-material/DarkMode';
+import Sun from '@mui/icons-material/LightMode';
+
+const ColorSchemePicker = () => {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+  if (!mounted) {
+    return null;
+  }
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+        } else {
+          setMode('light');
+        }
+      }}
+    >
```

Comment:
I just notice that this callback to handle the theme mode could be optimized, enhancing perfomance somehow?!

```suggestion
  const handleTheme = React.useCallback(() => {
    setMode(prevMode => prevMode === 'light' ? 'dark' : 'light');
  });

  return (
    <Button
      variant="outlined"
      onClick={handleTheme}
    >
```

I think this pattern is being used on every PR that add support for CSS variables.

Reply:
Sure, we could improve it 👍. Feel free to create a PR if you want. Note that these pages serves as a temporarily prove that the components work, we are anyway going to switch the docs to CSS variables soon so we would basically get rid of them.